packages/gestalt-codemods/24.0.0/button-replace-inline-fullWidth.js

```diff
@@ -0,0 +1,98 @@
+/*
+ * Converts
+ *  <Button /> to <Button fullWidth />
+ *  <Button inline/> to <Button />
+ *  <Button inline={false}/> to <Button fullWidth/>
+ *  <Button inline={inline} /> to CONSOLE.LOG for manual change />
+ */
+
+// yarn codemod --parser=flow -t=packages/gestalt-codemods/24.0.0/button-replace-inline-fullWidth.js relative/path/to/your/code
+
+const transformMap = {
+  inline: 'fullWidth',
+};
+
+export default function transformer(file, api) {
+  const j = api.jscodeshift;
+  const src = j(file.source);
+  let localIdentifierName;
+  let fileHasModifications = false;
+
+  src.find(j.ImportDeclaration).forEach((path) => {
+    const decl = path.node;
+    if (decl.source.value !== 'gestalt') {
+      return null;
+    }
+
+    localIdentifierName = decl.specifiers
+      .filter((node) => ['Button'].includes(node.imported.name))
+      .map((node) => node.local.name);
+    return null;
+  });
+
+  if (!localIdentifierName) {
+    return null;
+  }
+
+  const transform = src
+    .find(j.JSXElement)
+    .forEach((jsxElement) => {
+      const { node } = jsxElement;
+
+      if (!localIdentifierName.includes(node.openingElement.name.name)) {
+        return null;
+      }
+
+      const attrs = node.openingElement.attributes;
+
+      if (attrs.some((attr) => attr.type === 'JSXSpreadAttribute')) {
+        throw new Error(
+          `Remove Dynamic Text properties and rerun codemod. Location: ${file.path} @line: ${node.loc.start.line}`,
+        );
+      }
+
+      let addFullWidth = true;
+
+      const newAttrs = attrs
+        .map((attr) => {
+          const attrName = attr?.name?.name;
+          const disallowedProps = Object.keys(transformMap);
+
+          if (attrName && disallowedProps.includes(attrName)) {
+            addFullWidth = false;
+
+            if (attr.value === null) {
+              return false;
```

Comment:
nit: the point of these `return false` statements is to indicate that the attribute should be removed as part of the `.filter(Boolean)` sweep, right? I think returning null would better communicate that, as it typically is used to indicate "this is not a value". Booleans are typically valid values. This doesn't change the behavior at all since they're both falsy, but I think it makes the code a bit more readable.

Reply:
DONE